### ISSUE TITLE
Extreme Cpu Usage bug? 

### ISSUE THREAD

**Issue by theodoreippon** [Telegram-FOSS-Team/Telegram-FOSS#390]
Why is that? 

![Screenshot_20200224-033528_Device care](https://user-images.githubusercontent.com/25234246/75124608-1afdca80-56b9-11ea-8759-b59c5d72a0a7.jpg)

**Comment by ildar** [comment]
1. Did you do something special in Telegram, e.g. search, video etc.
2. Did you use the stock Telegram and see the same in it?

**Comment by theodoreippon** [comment]
Nope... As you can see i was on home screen... And before that, when the notification appeared i think i was on browser (brave browser if it matters).. 
No i didn't use the stock Telegram, i have a few weeks that i deleted it... 
I cannot think what it may triggered it, or why it used so much of my cpu in background... if it shows again I'll provide more details, but if it matters, a few days ago a notification appeared from android saying that telegram got my location in background, although i was not sharing my live location at the moment or had shared a current location moments before... 🤷‍♂️ 
Anyways, i hope that's a one time bug, or it can be fixed soon 😶 

Ohh my device is : Galaxy S9 on android 10

**Comment by damnfella** [comment]
That's most likely an issue with the way Samsung implements its "notifier" for background activities.

I don't know about location, but check this out:
https://telegram.org/faq#telegram-uses-the-camera-or-microphone-in-the-background

**Comment by c33s** [comment]
on one of my development phones, a samsung s4 mini (older version of android, for development and testing purposes), the cpu usage of telegram is really killing the battery. telegram is using 40-60% of the cpu. rooted lineage device with android 6.
rebooting sometimes help. sometimes i need to disable internet completly and force kill telegram.
a problem could be that the storeage is rather full. i often have to manually clear telegram media cache to not have a storage warning all the time.
currently it's using 45% cpu but my internet is disabled. logcat (using CatLog) filtered by `telegram` shows only a few entries which are from 10 minutes ago. if i don't filter by `telegram` i only get a lot of sensor and thermal (as my battery runs hot currently having 44.6°C) events.

how to find out what telegram is doing? is there a better alternative than CatLog? how can i see what is really going on?

**Comment by c33s** [comment]
maybe related to #436
